### DISCUSSION TITLE
Vercel.json not being parsed

### DISCUSSION THREAD

**Discussion by SID-dev-droid** [vercel/community#1925] [Help] [answered]
### Summary

Hi, I am trying to deploy to Vercel following a tutorial and all my code looks to be fine. It seems that when Vercel is building my app it isn't parsing my vercel.json file correctly as I don't see the Vite build command when no framework is chosen as my preset. When I do choose a framework such as Vite and manually override my build commands I get the front end of my application but my API can't connect for some reason can someone help if they know of a solution?

Thanks here is a link to my GitHub: https://github.com/Codename-SPYDER/booking-app.git

Running build in Washington, D.C., USA (East) – iad1
18:55:42.531 | Cloning github.com/Codename-SPYDER/booking-app (Branch: main, Commit: 7bb489e)
18:55:42.814 | Cloning completed: 282.523ms
18:55:42.856 | Previous build cache not available
18:55:42.964 | Running "vercel build"
18:55:43.533 | Vercel CLI 28.18.3
18:55:44.242 | Installing dependencies...
18:55:44.615 | yarn install v1.22.17
18:55:44.633 | warning package.json: No license field
18:55:44.677 | warning No license field
18:55:44.679 | [1/4] Resolving packages...
18:55:44.922 | [2/4] Fetching packages...
18:55:50.471 | [3/4] Linking dependencies...
18:55:56.322 | [4/4] Building fresh packages...
18:55:56.347 | success Saved lockfile.
18:55:56.351 | Done in 11.74s.
18:56:13.633 | Build Completed in /vercel/output [29s]
18:56:21.352 | Generated build outputs:
18:56:21.353 | - Static files: 36
18:56:21.353 | - Prerenders: 0
18:56:21.353 | - Serverless Functions: 4
18:56:21.353 | - Edge Functions: 0
18:56:21.353 | Serverless regions: Washington, D.C., USA
18:56:21.353 | Deployed outputs in 3s
18:56:21.650 | Build completed
18:56:38.468 | Uploading build cache [3.98 MB]...
18:56:39.355 | Build cache uploaded: 886.88ms

**Comment by amyegan** [comment, marked as answer]
Hi @SID-dev-droid! It looks you're using a monorepo structure. You should take a look at the [Root Directory](https://vercel.com/docs/concepts/deployments/configure-a-build#root-directory) setting. 

<img src="https://user-images.githubusercontent.com/8247105/229213799-501bece0-93d4-471d-a514-dc476a2cb2be.png" alt="Vercel project root directory setting" />
 
Please be reminded of Vercel’s [Terms of Service](https://vercel.com/legal/terms), [Trademark Policy](https://vercel.com/legal/trademark-policy), and [Copyright Policy](https://vercel.com/guides/how-does-vercel-handle-copyright-infringement-claims#dispute-resoluton) when working on projects that may be reported for violations.

**Reply by SID-dev-droid** [reply]
Hi @amyegan thanks for the reply, I just tried your suggestions, and the front end is deployed. However, my index.js doesn't seem to be receiving any of the requests or rewrites. I just receive a 404 error when I click something, even though my request path looks correct. Any other suggestions?

**Reply by amyegan** [reply]
That's because only the `client` app is being deployed. If you want to maintain the `api` as it's own project, then you can deploy it separately. Otherwise, you'll need to make some changes to your project structure to get it all deployed to the same project.

Some options to help you get it set up: 
- You could change the structure of the app so `api` and `src` are siblings at the root of your project with shared `package.json`. This will let the [Node.js Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#) automatically pick up the Serverless Functions in the `/api` directory.
- You could deploy the `client` and `api` as separate Vercel Projects, as is common when [using Monorepos](https://vercel.com/docs/concepts/monorepos#), and add [rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) or [CORS headers](https://vercel.com/guides/how-to-enable-cors) as needed to allow communication between them.
